### PR TITLE
Reduce cardinality of dynamo error messages

### PR DESCRIPTION
**Pull request by shitijagarwal**
Hello @sboora, @adrianlee44, 

Please review the following commits I made in branch 'shitij-errors-take2'.

0c9d837f5e249c71834f545526ae871ed7a3c929 (2016-03-02 14:56:12 -0800)
Reduce cardinality of dynamo error messages

R=@sboora
R=@adrianlee44

### CODE REVIEW

**Comment by shitijagarwal** [comment]
Not bumping the version yet, @adrianlee44 is making more changes to dynamite and will bump then.

File: lib/errors.js [hunk 1]
```diff
@@ -11,10 +11,12 @@ var util = require('util')
 function ConditionalError(data, msg) {
   Error.captureStackTrace(this)
   this.data = data
-  this.message = msg
+  this.message = 'The conditional request failed'
+  this.details = msg
+  this.table = data.TableName
 }
 util.inherits(ConditionalError, Error)
-ConditionalError.prototype.type = 'ConditionalError'
```
Comment (adrianlee44) [inline lib/errors.js]: This might be used. I would suggest creating a new `name` property.

Comment (shitijagarwal) [inline lib/errors.js]: Done.

**Comment by adrianlee44** [comment]
lgtm

**Comment by sboora** [comment]
LG!

**Pull request merged**